chore: add links to appropriate parser method/object (#296)
asyncapi/generator: docs/authoring.md

1. The default variables you have access to in any the template file are the following:
   - `asyncapi` that is a parsed spec file object. Read the [API](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument) of the Parser to understand to what structure you have access in this parameter.
   - `originalAsyncAPI` that is an original spec file before it is parsed. 
## File templates

It is possible to generate files for each specific object in your AsyncAPI documentation. 
For example, you can specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI. 
The following file-template names and extra variables in them are available:

>    - `$$channel$$`, within the template-file you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and [`channelName`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels). Where `channel` contains the current channel being rendered.

Reviewer: ```suggestion
   - `$$channel$$`, within the template-file you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and `channelName`. Where `channel` contains the current channel being rendered and `channelName` is the channel path to the corresponding channel. 
```
@Tenischev how about something like this? 🤔

Author: @jonaslagoni sorry, couldn't agree. The 

> `channelName` is the channel path to the corresponding channel

is super abstract for me. And also i doubt about "channel path" phrase, this could be just name of topic, name of queue e.t.c.

Author: What about following?
`$$channel$$` iterates over all channels. Within the template-file you have access to two variables `channel` and `channelName`, which are defined as `{% channel, channelName in asyncapi.`[`channels()`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels)` %}`.

Reviewer: IMO this gives no clarity to what `channelName` is, but for me it is pretty self-explanatory what channel path is when having read the specification, but yea since I am already somewhat involved with the parser and generator, it is really hard for me to say x, y, z is easier to understand as newcomers. So unless someone else weighs in i'd say you make the final call 😄

   - `$$securityScheme$$`, within the template-file you have access to two variables [`securityScheme`](https://github.com/asyncapi/parser-js/blob/master/API.md#SecurityScheme) and [`securitySchemeName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Components+securitySchemes). Where `securityScheme` contains the current security scheme being rendered.

The file name will be equal to `*Name` variable
##### Example
The file name is `$$schema$$.txt`, the content of this file is:
```
Schema name is '{{schemaName}}' and properties are:
{% for propName, prop in schema.properties() %}
- {{prop.uid()}}
{% endfor %}
```
With following AsyncAPI:
```
components:
  schemas: 
    peoplePayload:
      type: object
      properties:
        event:
          $ref: "#/components/schemas/people"
    people:
      type: object
      properties:
        id:
          type: integer
```

> Will generate two files `peoplePayload.txt` and `people.txt` with following content:

Reviewer 2: ```suggestion
The generator creates two files `peoplePayload.txt` and `people.txt` with the following content:
```

   - `asyncapi` that is a parsed spec file object. Read the [API](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument) of the Parser to understand to what structure you have access in this parameter.
   - `originalAsyncAPI` that is an original spec file before it is parsed. 
1. To separate functionality into files, you can specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI document. Such files templating gives you access to the following extra variables:
   - When using `$$channel$$` you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and [`channelName`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels). Where `channel` contains the current channel being rendered.
   - When using `$$message$$` you have access to two variables [`message`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message) and [`messageName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message+uid). Where `message` contains the current message being rendered.

>    - When using `$$schema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Only schemas from [Components object](https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-componentsobject-a-components-object) will be used. 

Reviewer 2: ```suggestion
   - When using `$$schema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Only schemas from [Components object](https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-componentsobject-a-components-object) are used. 
```

>    - When using `$$everySchema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Every [Schema object](https://www.asyncapi.com/docs/specifications/2.0.0/#schemaObject) from entire async api spec file will be used.

Reviewer 2: ```suggestion
   - When using `$$everySchema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Every [Schema object](https://www.asyncapi.com/docs/specifications/2.0.0/#schemaObject) from the entire AsyncAPI file is used.
```

> 1. To separate functionality into files, you can specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI document. Such files templating gives you access to the following extra variables:
>    - When using `$$channel$$` you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and [`channelName`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels). Where `channel` contains the current channel being rendered.
>    - When using `$$message$$` you have access to two variables [`message`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message) and [`messageName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message+uid). Where `message` contains the current message being rendered.
>    - When using `$$schema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Only schemas from [Components object](https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-componentsobject-a-components-object) are used. 
>    - When using `$$everySchema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Every [Schema object](https://www.asyncapi.com/docs/specifications/2.0.0/#schemaObject) from the entire AsyncAPI file is used.
>    - When using `$$parameter$$` you have access to two variables [`parameter`](https://github.com/asyncapi/parser-js/blob/master/API.md#ChannelParameter) and [`parameterName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel+parameters). Where `parameter` contains the current parameter being rendered.
>    - When using `$$securityScheme$$` you have access to two variables [`securityScheme`](https://github.com/asyncapi/parser-js/blob/master/API.md#SecurityScheme) and [`securitySchemeName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Components+securitySchemes). Where `securityScheme` contains the current security scheme being rendered.

Reviewer: Honestly I am not the biggest fan of it being listed under `Common assumptions`  and since we are changing it how about we make the functionality a headline such as filters, Hooks has. 

```suggestion
## Separate functionality into files

The generator supports multiple separation of components, schemas, messages etc. I.e specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI document. Below is a list of all the file templates and their corresponding global variable which are available.

- When using `$$channel$$` you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and [`channelName`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels). Where `channel` contains the current channel being rendered.
- When using `$$message$$` you have access to two variables [`message`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message) and [`messageName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Message+uid). Where `message` contains the current message being rendered. 
- When using `$$schema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Only schemas from [Components object](https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-componentsobject-a-components-object) are used.  
- When using `$$everySchema$$` you have access to two variables [`schema`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema) and [`schemaName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Schema+uid). Where `schema` contains the current schema being rendered. Every [Schema object](https://www.asyncapi.com/docs/specifications/2.0.0/#schemaObject) from the entire AsyncAPI file is used. 
- When using `$$parameter$$` you have access to two variables [`parameter`](https://github.com/asyncapi/parser-js/blob/master/API.md#ChannelParameter) and [`parameterName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel+parameters). Where `parameter` contains the current parameter being rendered.
- When using `$$securityScheme$$` you have access to two variables [`securityScheme`](https://github.com/asyncapi/parser-js/blob/master/API.md#SecurityScheme) and [`securitySchemeName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Components+securitySchemes). Where `securityScheme` contains the current security scheme being rendered.
```
What do you guys think ?

Author: in new version

>    - When using `$$securityScheme$$` you have access to two variables [`securityScheme`](https://github.com/asyncapi/parser-js/blob/master/API.md#SecurityScheme) and [`securitySchemeName`](https://github.com/asyncapi/parser-js/blob/master/API.md#Components+securitySchemes). Where `securityScheme` contains the current security scheme being rendered.

Reviewer: Further more for securityScheme and channel, when the name links me to the https://github.com/asyncapi/parser-js/blob/master/API.md#Components+securitySchemes and https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels I would expect I would have access to the map instead of just the key (which is what is provided) maybe add some extra text for those 🤔 ?

Author: I understand your concern, but to the pity we do not have a direct way in API to get only the name of channel, channel parameter or security schema. We had them only in the list of Object.<string, Type>.
For me there is a thread of between simplicity, compactness of explanation and ton of hints in bracers.
If you have some good sentence with explanation in mind, please suggest.

Reviewer: Left a new review with suggestion 😄
